### ISSUE TITLE
NSIS - Option to show installation details using ShowInstDetails

### ISSUE THREAD

**Issue by yafp** [electron-userland/electron-builder#4719]
* **Version**:  22.3.2

* **Target**:  windows

I would like to use the switch `ShowInstDetails` to show details about what the installer is currently doing.
Is there a way to use this parameter already or would it be possible to add support for it?

Why: It would give feedback to the user about the progress and details

**Comment by yafp** [comment]
could that be realized using `include`  ?

**Comment by yafp** [comment]
Ok, what i did is

**1. installer.nsh**
Created an installer.nsh file with the following content

```
ShowInstDetails show
ShowUninstDetails show
```

**2. Package.json**
In my package.json i am referencing this installer.nsh script as follows:

```
"nsis": {
            "include": "resources/installer/nsis/installer.nsh",
            "deleteAppDataOnUninstall": true,
            "unicode": false,
            "artifactName": "${productName}-${version}-win-installer-${arch}.${ext}",
            "createDesktopShortcut": true,
            "createStartMenuShortcut": true,
            "oneClick": false,
            "allowElevation": true,
            "allowToChangeInstallationDirectory": true,
            "runAfterFinish": false
        },
```


Building throws no errors, but it seems to have no effect either.
Any idea what i am doing wrong?

**Comment by LeoBali** [comment]
Just add it to custom header section

**Comment by yafp** [comment]
@LeoBali 
thanks for the info.
My installer is now in fact showing a log-kind element where details could be displayed - but the field is basically empty.

So adding your code to my installer.nsh does force a ui element for installation / uninstallation informations - but it not automatically provide informations about the actual process.

Is that the expected behaviour?

**Comment by yafp** [comment]
Yes i would still like to know how to show details about the installation and uninstall process

**Comment by AxelTerizaki** [comment]
Thanks for the tip, I get the same behavior as I'm trying to print out installation log/details using that. I've been looking for a way to find installer logs (see #4918)

**Comment by keating99** [comment]
I am getting the same issue with my installer. Is there any more information on this?

**Comment by sstopkin** [comment]
I'm getting same issue with version 22.11.7.
Trying to build - "oneClick": false

As I can see - "ShowInstDetails" and "ShowUninstDetails" options are set to "nevershow". But for some reason overriding works fine for ShowUninstDetails and doesn't work for ShowInstDetails.
https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/templates/nsis/common.nsh#L5
As @yafp mentioned - I'm getting an empty UI from but no any installation log inside.

Any idea what happened here? Thanks

**Comment by LightNeverDies** [comment]
Is there any solution for it?

**Comment by zhouJiecode** [comment]
Is there any solution for it?

**Comment by zhouJiecode** [comment]
i known the reason: look at the code in `node_modules/app-builder-lib/templates/nsis/installSection.nsh`:
```javascript
${IfNot} ${Silent}
  SetDetailsPrint none
${endif}
```

we have to set SetDetailsPrint to both.

what i do is:
- copy the file to my directory, modify SetDetailsPrint to both: `SetDetailsPrint both`
- add a beforePack hook in `package.json`: `"beforePack": "./script/beforePack.js",`， and in the hook, to cover `node_modules/app-builder-lib/templates/nsis/installSection.nsh` with my file
```javascript
// beforePack.js
exports.default = async function () {
  const newInstallNshPath = path.resolve(__dirname, '../build/myInstallSection.nsh');
  const destNshPath = path.resolve(__dirname, '../node_modules/app-builder-lib/templates/nsis/installSection.nsh');
  fs.renameSync(newInstallNshPath, destNshPath);
}
```

**Comment by zhouJiecode** [comment]
i known the reason: look at the code in `node_modules/app-builder-lib/templates/nsis/installSection.nsh`:
```javascript
${IfNot} ${Silent}
  SetDetailsPrint none
${endif}
```

we have to set SetDetailsPrint to both.

what i do is:
- copy the file to my directory, modify SetDetailsPrint to both: `SetDetailsPrint both`
- add a beforePack hook in `package.json`: `"beforePack": "./script/beforePack.js",`， and in the hook, to cover `node_modules/app-builder-lib/templates/nsis/installSection.nsh` with my file
```javascript
// beforePack.js
exports.default = async function () {
  const newInstallNshPath = path.resolve(__dirname, '../build/myInstallSection.nsh');
  const destNshPath = path.resolve(__dirname, '../node_modules/app-builder-lib/templates/nsis/installSection.nsh');
  fs.renameSync(newInstallNshPath, destNshPath);
}
```

these code is needed too, they can be put in your `installer.nsh`:
```javascript
!macro customHeader
  ShowInstDetails show
  ShowUninstDetails show
!macroend
```

**Comment by mqliutie** [comment]
> i known the reason: look at the code in `node_modules/app-builder-lib/templates/nsis/installSection.nsh`:
> 
> ```js
> ${IfNot} ${Silent}
>   SetDetailsPrint none
> ${endif}
> ```
> 
> we have to set SetDetailsPrint to both.
> 
> what i do is:
> 
> * copy the file to my directory, modify SetDetailsPrint to both: `SetDetailsPrint both`
> * add a beforePack hook in `package.json`: `"beforePack": "./script/beforePack.js",`， and in the hook, to cover `node_modules/app-builder-lib/templates/nsis/installSection.nsh` with my file
> 
> ```js
> // beforePack.js
> exports.default = async function () {
>   const newInstallNshPath = path.resolve(__dirname, '../build/myInstallSection.nsh');
>   const destNshPath = path.resolve(__dirname, '../node_modules/app-builder-lib/templates/nsis/installSection.nsh');
>   fs.renameSync(newInstallNshPath, destNshPath);
> }
> ```
> 
> these code is needed too, they can be put in your `installer.nsh`:
> 
> ```js
> !macro customHeader
>   ShowInstDetails show
>   ShowUninstDetails show
> !macroend
> ```

Good job
